view/src/bin/pviewd.rs

```diff
@@ -34,6 +36,49 @@ struct Opt {
     pd_port: u16,
 }
 
+impl Opt {
+    #[instrument(skip(self))]
+    pub async fn status(&self) -> Result<StatusResponse, anyhow::Error> {
```

Comment:
We'll need to get the status from inside of the `ViewService` while processing the status request, so we need the code that fetches the Tendermint status to be in the `ViewService`.

We should probably change `ViewService::new` to take `node` and `tendermint_port` and store them internally, so we can put this code in a
```rust
impl ViewService {
    /// Returns the `latest_block_height` and `catching_up` fields from the full node we're syncing from.
    async fn node_block_height(&self) -> Result<(u64, bool)> { ... }
}
```
that we can call from inside the `ViewService`.

Reply:
Cool, will fix!